apps/admin-gui/src/app/facilities/pages/resource-detail-page/resource-assigned-members/resource-assigned-members.component.ts

```diff
@@ -35,21 +62,74 @@ export class ResourceAssignedMembersComponent implements OnInit {
   refreshTable(): void {
     this.loading = true;
     this.resourceService.getAssignedRichMembers(this.resource.id).subscribe((members) => {
-      this.members = members;
-      this.setAuthRights();
-      this.loading = false;
+      this.setAuthRights(members);
+      this.consentService.getConsentHubByResource(this.resource.id).subscribe((hub) => {
+        // include consents if they are turned on and enforced
+        this.includeConsents =
+          this.storeService.getProperty('enforce_consents') && hub.enforceConsents;
+        if (this.includeConsents) {
+          this.columns = ['id', 'fullName', 'consentStatus'];
+          this.consentService
+            .getConsentsForConsentHubByResource(this.resource.id)
+            .subscribe((consents) => {
+              this.consents = consents;
+              this.members = this.getConsentsForMembers(members).filter(
+                (member) =>
+                  this.selectedConsentStatuses.length === 0 ||
+                  this.selectedConsentStatuses.includes(member.consent.toUpperCase())
+              );
+            });
+        } else {
+          this.members = members as MemberWithConsentStatus[];
+        }
+        this.loading = false;
+      });
     });
   }
 
-  setAuthRights(): void {
-    if (this.members.length !== 0) {
-      this.routeAuth = this.authResolver.isAuthorized('getMemberById_int_policy', [
-        this.members[0],
-      ]);
+  setAuthRights(members: RichMember[]): void {
+    if (members.length !== 0) {
+      this.routeAuth = this.authResolver.isAuthorized('getMemberById_int_policy', [members[0]]);
     }
   }
 
   applyFilter(filterValue: string): void {
     this.filterValue = filterValue;
   }
+
+  getConsentsForMembers(members: RichMember[]): MemberWithConsentStatus[] {
+    const result: MemberWithConsentStatus[] = [];
+    members.forEach((member) => {
+      const mwc: MemberWithConsentStatus = member;
+      mwc.consent = this.translate.instant(
+        'CONSENTS.STATUS_' + this.consentStatusPipe.transform(member.userId, this.consents)
+      ) as string;
```

Comment:
I would suggest keeping the raw consent status inside of the object. Also for translation use the `PerunTranslateService`.

Reply:
With this change, the data is exported without the consent status translated (which is an inconsistency compared to users dynamic list where it's translated, should we not translate it there as well?)

Reply:
If we want the consents to be translated when exported (seems reasonable) we could can do that it the `getExportDataForColumn`.

Reply:
Good point. I overlooked it. However, there is a problem with using injected translate service in the `getExportDataForColumn` static method. This method is used only once in `getExportDataForColumnFun` in the same component. Is it necessary for this method to be static?

Reply:
Yep, that's what I was dealing with when implementing the consent column for the assigned users page. I believe there was a reason, cannot remember though. @bodnara ?